utils/UniswapPoolsHelpers.ts

```diff
@@ -110,9 +110,131 @@ export async function getPools(hre: HardhatRuntimeEnvironment): Promise<{ name:
     pools.push([data0, data1]);
     registeredPools.add(poolKey);
   }
-
   const supportedLiquidityPools = getListOfSupportedLiquidityPools(hre.network.name);
   supportedLiquidityPools.forEach(({ tokenA, tokenB }) => registerPool(tokenToPoolTokenData[tokenA], tokenToPoolTokenData[tokenB]));
-
   return pools;
 }
+
+export async function getAvailableSwapLinks(hre: HardhatRuntimeEnvironment) {
+  const pools = await getPools(hre);
+
+  const availableLinks = [];
+
+  for (const pool of pools) {
+    availableLinks.push({ from: pool[0].token.address, to: pool[1].token.address });
+    availableLinks.push({ from: pool[1].token.address, to: pool[0].token.address });
+  }
+
+  return availableLinks;
+}
+
+export async function generatePaths(
+  hre: HardhatRuntimeEnvironment,
+  amountIn: BigNumber,
+  addressIn: string,
+  addressOut: string
+): Promise<{ path: string[]; amountOut: BigNumber }[]> {
+  const availableSwapLinks = await getAvailableSwapLinks(hre);
+
+  const midTokens = [];
+
+  for (const link1 of availableSwapLinks) {
+    if (link1.from !== addressIn) {
+      continue;
+    }
+    for (const link2 of availableSwapLinks) {
+      if (link1.to !== link2.from) {
+        continue;
+      }
+      if (link2.to !== addressOut) {
+        continue;
+      }
+
+      midTokens.push(link1.to);
+    }
+  }
+
+  const paths = [[addressIn, addressOut], ...midTokens.map(x => [addressIn, x, addressOut])];
+
+  const router = await getUniswapRouter(hre);
+  const pathsPrices = [];
+
+  for (const path of paths) {
+    let amountsOut;
+    try {
+      amountsOut = await router.getAmountsOut(amountIn, path);
+    } catch {
+      continue; // handle unsupported paths like [wrbtc -> eths -> bdx]
+    }
+
+    pathsPrices.push({
+      path: path,
+      amountOut: amountsOut[amountsOut.length - 1]
+    });
+  }
+
+  return pathsPrices;
+}
+
+export async function generateAllPaths(
+  hre: HardhatRuntimeEnvironment,
+  excludeAsInputTokens: string[],
+  requiredTokensOut: string[]
+): Promise<string[][]> {
+  const allTokensAddresses = await getAllTokensAddresses(hre, excludeAsInputTokens);
+  let allRequiredPaths: string[][] = [];
+  for (const tokenAddressIn of allTokensAddresses) {
+    for (const requiredTokenOut of requiredTokensOut) {
+      if (requiredTokensOut.includes(tokenAddressIn)) {
+        continue;
+      }
+      const pathsWithAmountOut = await generatePaths(hre, to_d18(1), tokenAddressIn, requiredTokenOut);
```

Comment:
This problem probably has existed for a while, but we should probably change `to_d18(1)` to `to_d18(1e-5)`. 1 BTC is a lot and has a big price impact. WDYT @kazazor @tkeidar